### PR TITLE
Audiobieldev.walink.org

### PR DESCRIPTION
**Pull request by wkelly17**
## Move code for sending messages to service bus to prod. 

I'm sorry for the excessive commits in here for what is not really all that much code.  I was essentially remote debugging through a production cycle of deploy to jenkins / run / wait / check.  But here's the high level.

1. dockerstack/run.sh -> grab env vars from OP. adjust how the project is run in Jenkins. 
2. At the end of each worker cycle, look through the cdn, batch messages together based on language/project, and send those off to azure service bus
3. Can exclude files bassed on runtime args https://github.com/Bible-Translation-Tools/Fetcher/compare/audio.bibleineverylanguage.org...audiobieldev.walink.org#diff-261713a8315e42b6d65bb74407b512493386d4fa3670733a61cf430bb32e8a5dR213-R215.  My runs so far excluded verses, lo quality mp3s, and TR files based on previous conversations. 

@danparisd  -> We'll need to add and adjust some stuff in jenkins.  I can do it, or you can. Just lmk

## Before merging:
* [ ] If this PR targets dev branch, do a Squash-merge.
* [ ] If the target branch is prod, do a Regular merge.

- - -
This change is [Reviewable](https://reviewable.io/reviews/Bible-Translation-Tools/Fetcher/212)

### CODE REVIEW

**Review by mXaln — DISMISSED** [review]
LGTM

File: fetcher-pipeline/app.py [hunk 1]
```diff
@@ -74,17 +82,143 @@ def get_report(reports):
...
+                return [in_array[i:i + size] for i in range(0, len(in_array), size)]
+
+            with open("./book_catalog.json", "r") as json_file:
+                books = json.load(json_file)
+
 
+            # iterate through each language
+            logging.debug(f"iterating through {self.__ftp_dir}")
+            for language_dir in self.__ftp_dir.iterdir():
+                if language_dir.is_dir() or "analysis" not in language_dir:
```
Comment (PurpleGuitar) [inline fetcher-pipeline/app.py]: Generally, when you have the situation where you're in a loop and you only want to execute if certain things are true, it's more readable to "fail-fast": test for the negative and skip the loop if it doesn't match.  It also allows you to keep your indenting much shallower.

So in this case:

```python
for language...:

    if not (langauge_dir.is_dir() and "analysis" not in language_dir):
        continue 

    logging.debug(...
```

File: fetcher-pipeline/app.py [hunk 2]
```diff
@@ -74,17 +82,143 @@ def get_report(reports):
...
+                    logging.debug(f"doing {language_dir}")
+                    for project_dir in language_dir.iterdir():
+                        if project_dir.is_dir():
```
Comment (PurpleGuitar) [inline fetcher-pipeline/app.py]: Similarly:

```python
for project_dir in language_dir.iterdir():
    if not project_dir.is_dir():
        continue
    logging.debug(...
```

File: fetcher-pipeline/app.py [hunk 3]
```diff
@@ -74,17 +82,143 @@ def get_report(reports):
...
+               
+                logging.debug(f"Done sending messages to queue. Sent {len(messages)} messages.")
+    
+    @staticmethod
+    def should_skip_file(file_path: Path, exlude_args: List[str]) -> bool:
+        should_skip = False
+        for arg in exlude_args:
+            if arg.startswith("."):
+                if file_path.suffix == arg:
+                    should_skip = True
```
Comment (PurpleGuitar) [inline fetcher-pipeline/app.py]: It would be more efficient to just `return True` here and in the other branches.  Then you can skip allocating `should_skip` and just `return False` at the end of the function.

**Review by PurpleGuitar — DISMISSED** [review]

**Review by danparisd — APPROVED** [review]
LGTM

**Review by AnonymousWalker — APPROVED** [review]
